prov/sockets/include/sock.h

```diff
@@ -1211,6 +1218,14 @@ int sock_epoll_wait(struct sock_epoll_set *set, int timeout);
 int sock_epoll_get_fd_at_index(struct sock_epoll_set *set, int index);
 void sock_epoll_close(struct sock_epoll_set *set);
 
+int sock_epoll_map_init(struct sock_epoll_map *emap);
+int sock_epoll_map_destroy(struct sock_epoll_map *emap);
+int sock_epoll_map_add(struct sock_epoll_map *emap, int fd, void *ptr);
+int sock_epoll_map_del(struct sock_epoll_map *emap, int fd);
+void *sock_epoll_map_lookup(struct sock_epoll_map *emap, int fd);
+int sock_epoll_map_signal(struct sock_epoll_map *emap);
+int sock_epoll_map_wait(struct sock_epoll_map *emap);
```

Comment:
You are right, the fi_epoll abstraction already provides the functionality I need for the epoll set.
I created a new PR (see PR #3221) that extracts the patches that are not directly related to the new connection manager for the sockets provider.
In this new PR, I have extended the fi_epoll_wait() call to allow multiple events to be returned.
I have also deprecated the the sock_epoll_* interface in favor of the common fi_epoll_* interface in the sockets provider.